packages/components/src/components/DistributionChart.tsx

```diff
@@ -75,10 +102,33 @@ export const DistributionChart: React.FC<DistributionChartProps> = (props) => {
       );
       widthProp = 20;
     }
-    const domain = shapes.value.flatMap((shape) =>
+    const predomain = shapes.value.flatMap((shape) =>
       shape.discrete.concat(shape.continuous)
     );
 
+    const domain =
+      xAxis === "dateTime"
+        ? predomain.map((p) => ({ dateTime: p.x, y: p.y }))
+        : predomain;
+
+    const data =
+      xAxis === "dateTime"
+        ? shapes.value.map((val) => {
+            return {
+              ...val,
+              continuous: val.continuous.map((p) => {
+                return { dateTime: p.x, y: p.y };
```

Comment:
Why do you rename the field to dateTime? This creates a lot of unnecessary code below

Reply:
Great question, haha, you're totally right. I think I just got confused about some Vega feature while debugging and this was left in. Amending now.